nncf/experimental/openvino_native/graph/nncf_graph_builder.py

```diff
@@ -0,0 +1,114 @@
+"""
+ Copyright (c) 2022 Intel Corporation
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+      http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+
+import openvino.runtime as ov
+
+from nncf.common.graph import NNCFGraph
+from nncf.common.graph.layer_attributes import Dtype
+from nncf.common.graph.operator_metatypes import UnknownMetatype
+from nncf.common.utils.logger import logger as nncf_logger
+
+from nncf.experimental.openvino_native.graph.metatypes.openvino_metatypes import OV_OPERATION_METATYPES
+
+
+class GraphConverter:
+    """
+    Builds the NNCFGraph from an OpenVINO model.
+    """
+
+    @staticmethod
+    def _is_valid_openvino_metatype(node: ov.Node) -> bool:
+        """
+        Checks whether the node has the metatype which should be added to the NNCFGraph.
+
+        :param node: Node to be checked.
+        :return: True if the metatype is valid and False if not.
+        """
+        node_type = node.get_type_name()
+        metatype = OV_OPERATION_METATYPES.get_operator_metatype_by_op_name(node_type)
+        if metatype == UnknownMetatype:
+            node_name = node.get_friendly_name()
+            nncf_logger.warning(
+                'The node with name {} with type {} was mapped to UnknownMetatype,'
+                ' which means that there was not registered such NNCF metatype. '
+                'Please, Inform the NNCF developers about this message.'.format(
+                    node_name, node_type))
+        return True
+
+    @staticmethod
+    def convert_ov_dtype_to_nncf_dtype(ov_dtype: str) -> Dtype:
+        """
+        Converts the primitive types from the OpenVINO domain to the NNCF domain.
+
+        :param ov_dtype: OpenVINO primitive typename.
+        :return: NNCF primitive type.
+        """
+        conversion_map = {
+            'f16': 'float',
+            'f32': 'float',
+            'i4': 'int',
+            'i8': 'int',
+            'i32': 'int',
+            'i64': 'int',
+            'u1': 'int',
+            'u4': 'int',
+            'u8': 'int',
+            'u32': 'int',
+            'u64': 'int',
+        }
+        if ov_dtype not in conversion_map:
+            raise NotImplementedError(f'NNCF is not yet supported OpenVINO data type: {ov_dtype}.')
+        return Dtype(conversion_map[ov_dtype])
+
+    @staticmethod
+    def create_nncf_graph(model: ov.Model) -> NNCFGraph:
+        """
```

Comment:
For correct quantizer placement in other backends we add special Input and Output nodes. It is not necessary for now, bit you should keep it in mind